site/pages/api/type-of-consequence.api.ts

```diff
@@ -12,14 +12,14 @@ import { flattenZodErrors } from "../../utils";
 import {
     destroyCookieOnResponseObject,
     isDisruptionFromTemplate,
+    redirectTo,
     redirectToError,
     redirectToWithQueryParams,
     setCookieOnResponseObject,
 } from "../../utils/apiUtils";
 
 const addConsequence = (req: NextApiRequest, res: NextApiResponse): void => {
     try {
-        const { template } = req.query;
         const isFromTemplate = isDisruptionFromTemplate(req);
```

Comment:
I'd suggest getting isFromTemplate by passing it through to the api as a req.query:

```const { isFromTemplate } = req.query```

The isDisruptionFromTemplate function is a bit weird as it uses the req.headers.referer when it would be better to use req.query

Reply:
done